devlib/module/cgroups.py

```diff
@@ -32,157 +30,230 @@ def __new__(cls, arg):
-        raise NotImplementedError()
-
-    def mount(self, device, mount_root):
-        self.target = device
-        self.mount_point = device.path.join(mount_root, self.mount_name)
-        mounted = self.target.list_file_systems()
-        if self.mount_point in [e.mount_point for e in mounted]:
-            self.logger.debug('controller is already mounted.')
+        try:
+            exists = target.execute(
+                    'grep {} /proc/cgroups'.format(self.kind))
+        except TargetError:
+            return False
+        return True
+
+    def mount(self, target, mount_root):
+
+        mounted = target.list_file_systems()
+        if self.mount_name in [e.device for e in mounted]:
+            # Identify mount point if controller is already in use
+            self.mount_point = [
+                    fs.mount_point
+                    for fs in mounted
+                    if fs.device == self.mount_name
+                ][0]
         else:
-            self.target.execute('mkdir -p {} 2>/dev/null'.format(self.mount_point),
-                                as_root=True)
-            self.target.execute('mount -t cgroup -o {} {} {}'.format(self.kind,
-                                                                     self.mount_name,
-                                                                     self.mount_point),
-                                as_root=True)
+            # Mount the controller if not already in use
+            self.mount_point = target.path.join(mount_root, self.mount_name)
+            target.execute('mkdir -p {} 2>/dev/null'\
+                    .format(self.mount_point), as_root=True)
+            target.execute('mount -t cgroup -o {} {} {}'\
+                    .format(self.kind,
+                            self.mount_name,
+                            self.mount_point),
+                            as_root=True)
+
+        self.logger.info('Controller %s mounted under: %s',
+            self.kind, self.mount_point)
+
+        # Mark this contoller as available
+        self.target = target
+
+        # Create root control group
+        self.cgroup('/')
+
+    def cgroup(self, name):
+        if not self.target:
+            raise RuntimeError('CGroup creation failed: {} controller not mounted'\
+                    .format(self.kind))
+        if name not in self._cgroups:
+            self._cgroups[name] = CGroup(self, name)
+        return self._cgroups[name]
+
+    def exists(self, name):
+        if not self.target:
+            raise RuntimeError('CGroup creation failed: {} controller not mounted'\
+                    .format(self.kind))
+        if name not in self._cgroups:
+            self._cgroups[name] = CGroup(self, name, create=False)
+        return self._cgroups[name].existe()
+
+    def list_all(self):
+        self.logger.debug('Listing groups for %s controller', self.kind)
+        output = self.target.execute('find {} -type d'.format(self.mount_point))
```

Comment:
You mean something like:
self.target.execute('{} find {} -type d'.format(self.target.busybox, self.mount_point))?
... going to update.